### DISCUSSION TITLE
UML Designer?

### DISCUSSION THREAD

**Discussion by mikes2nd** [Xomega-Net/Xomega.Net4VS#19] [Q&A]
Do you recommend a UML designer like Violet, UMLet, Gaphor etc to do our design and generate a database?

Thanks, 
Mike

**Comment by XomegaNet** [comment]
Hi Mike, thanks for your interest in Xomega.

You can use any graphical designer to create UML or ER diagrams and generate a database schema from it. We don't have recommendations for any particular product, as long as it's capable of doing a good job.

If you are designing a new app from scratch, would you consider a model-first approach where you'd define Xomega objects first, which then allows you to generate EF Core entities and the DB schema? This way you can start by using rich logical types, and model nested sub-objects during your design. If you'd like to see a graphical view for the model entities, you can also generate an Entity Domain Model (EDM) from it and create multiple EDM diagrams for specific modules.

Also, if Xomega had its own graphical designer for modeling domain entities and services, how valuable do you think you'd find it?

**Comment by mikes2nd** [comment]
That would be great but maybe a big lift, or use AI :)  Where I can start with a basic framework for say a single page web app with MS/Google/FB/etc login authorization with basic roles and various other application types. Then add objects by voice or typing...   Say "add multiple object vehicle which relates to single owner object"... :)   Or "Create object Vehicles which relates to owner"...  Functional app style...  "add a string column named purchase price to object vehicles"...

**Reply by XomegaNet** [reply]
Right, Gen AI sounds perfect for it, although it's probably also a big lift. The trick is to allow it to modify the existing model for iterative development, and make it always produce valid and appropriate output. Maybe a Copilot-style would work best in this case. Or some AI-assisted wizards.

For green-field development, what do you think of being able to add some preconfigured standard modules from some collection or a marketplace? For example, to add users, roles and privileges you'd click Add New Item and select an Identity module, which would create all those objects. Then you'd be able to remove fields you don't want or add new ones as needed. The same could be done for some other common modules, such as customers, payments, etc. Just wondering how useful you would find it...

**Comment by mikes2nd** [comment]
I need to research the interaction between a nuget package and migration... there is a facebook identity nuget package as you know.

Some common 'business objects" would be interesting though, similar to a nuget package...  Like customers, items, orders, invoices, addresses, products, photos etc would be interesting.  They would obviously be very simple and USA/EU "common" as much as possible.
